Created PUT api for admin page

RYin-28 opened the pull request:
## Issue Name
Created api and hook for editing products
## Describe your changes
- setting json changed b/c of extensions
- Edited 'EditProductForm' to update table when product is created
- Edited subsequent 'use-add'product' to reflect previous point
- Created 'use-edit-product' hook
- Cleaned up 'use-get-product'
- Changed form submit operation to accommodate both new product creation and editing old products
- Added console log to view event
- Created PUT API
- Created PUT route
## Checklist before requesting a review

- [X] I have performed a self-review of my code
- [X] If it is a core feature, I have added thorough tests.
- [X] Do we need to implement analytics?

RYin-28 committed:
Created PUT api for admin page
- setting json changed b/c of extensions
- Edited 'EditProductForm' to update table when product is created
- Edited subsequent 'use-add'product' to reflect previous point
- Created 'use-edit-product' hook
- Cleaned up 'use-get-product'
- Changed form submit operation to accommodate both new product creation and editing old products
- Added console log to view event
- Created PUT API
- Created PUT route

drmicrochip97 commented on .vscode/settings.json:
```diff
@@ -13,5 +13,17 @@
   },
-  "eslint.validate": ["json"]
+  "eslint.validate": ["json"],
+  "cSpell.words": ["mantine"],
```
what's cSpell? if its the spell check we can remove this

RYin-28 replied:
That line is for words I add to the 'dictionary' that Spell Checker didn't recognize. I'll delete it and remove the extension since it wasn't too impactful for me.

drmicrochip97 commented on src/app/admin/edit-product-form/index.tsx:
```diff
@@ -52,16 +56,24 @@ export default function EditProductForm({ product, getProducts, onClose }: Props
+                await editProduct(product.id, editedProductData)
+              } else {
+                console.log('making new product')
+                createProduct(form.values)
```
you can remove these log statements

drmicrochip97 commented on src/app/admin/edit-product-form/index.tsx:
```diff
@@ -144,8 +156,8 @@ export default function EditProductForm({ product, getProducts, onClose }: Props
-                text={editing ? 'Update' : 'Submit'}
-                onClick={() => (editing ? console.log('Editing') : console.log('Creating'))}
+                text={product ? 'Update' : 'Submit'}
+                onClick={() => (product ? console.log('Editing') : console.log('Creating'))}
```
is this onClick necessary, if not remove

RYin-28 replied:
onClick is necessary since it's defined in the component ()
`interface Props {text: string, onClick: () => void}`

`export function RoundButton({ text, onClick }: Props)`

drmicrochip97 commented on src/app/admin/products-table/products-table.tsx:
```diff
@@ -72,6 +72,7 @@ export function ProductsTable({ products, openEdit, getProducts }: Props) {
             <RoundButton
               text="Yes"
               onClick={() => {
+                console.log('Deleting:', productToDelete)
```
remove deleting log

drmicrochip97 commented on src/app/admin/use-edit-product.ts:
```diff
@@ -0,0 +1,20 @@
+  const editProduct = useCallback(
+    async (id: string, updatedProduct: ProductAttr) => {
+      try {
+        console.log('udpate prod:', updatedProduct)
```
remove log

drmicrochip97 commented on src/app/admin/use-edit-product.ts:
```diff
@@ -0,0 +1,20 @@
+        await req('PUT /product/[id]', { params: { id }, data: updatedProduct })
+        await getProducts()
+      } catch (error) {
+        console.log('Error in editProduct:', error)
```
use console.error

drmicrochip97 commented on src/app/api/product/[id]/route.ts:
```diff
@@ -10,3 +11,24 @@ export async function DELETE(_: Request, { params }: { params: Params<'DELETE /p
+
+export async function PUT(request: Request, { params }: { params: Params<'PUT /product/[id]'>; body: ProductAttr }) {
+  const body: Body<'PUT /product/[id]'> = await request.json()
+  console.log('Body: ', body)
```
remove log statements

drmicrochip97 commented on src/app/api/product/[id]/route.ts:
```diff
@@ -10,3 +11,24 @@ export async function DELETE(_: Request, { params }: { params: Params<'DELETE /p
+      price: price !== undefined ? { set: price } : undefined,
+      type: type !== undefined ? { set: type } : undefined,
+      families: families !== undefined ? { set: families } : undefined,
+      description: description !== undefined ? { set: description } : undefined,
```
this while undefined checking seems is unnecessary. refactor and simplify

RYin-28 committed:
[Commit Changes]
- Removed spell checker extension and json settings
- Removed console logs
- Removed console logs from onClick action
- Simplified PUT route body
